From d47dcd44ea388c387e44d09ba450b81c1a068ff4 Mon Sep 17 00:00:00 2001
From: Simon Busch <morphis@gravedo.de>
Date: Sat, 7 Dec 2013 17:26:57 +0100
Subject: [PATCH] Retrieve WIFIoADDR/BToADDR properties also from nyx device
 info module

Both properties WIFIoADDR and BToADDR were stored as tokens before. We're offering support
to also retrieve the same information through the device info module for quite some time
so we can also add support for both names. This is also backward compatible with existing
implementations providing both as tokens as tokens always have a higher preference then
retrieving the values from elsewhere.

Open-webOS-DCO-1.0-Signed-off-by: Simon Busch <morphis@gravedo.de>
---
 libluna-prefs/lunaprefs.c | 18 +++++++++++++++++-
 1 file changed, 17 insertions(+), 1 deletion(-)

diff --git a/libluna-prefs/lunaprefs.c b/libluna-prefs/lunaprefs.c
index c88da49..cbaea29 100644
--- a/libluna-prefs/lunaprefs.c
+++ b/libluna-prefs/lunaprefs.c
@@ -102,7 +102,8 @@
 #define PROP_NAME_FREESPACE       "storageFreeSpace"
 #define PROP_NAME_PREVPANIC       "prevBootPanicked"
 #define PROP_NAME_PREVSHUTCLEAN   "prevShutdownClean"
-
+#define PROP_NAME_WIFIOADDR       "WIFIoADDR"
+#define PROP_NAME_BTOADDR         "BToADDR"
 
 static const char* PALM_TOKEN_PREFIX = "com.palm.properties.";
 
@@ -824,6 +825,15 @@ static LPErr read_machine_type(char** jstr,const char* key)
             {
                 error = nyx_device_info_query(device, NYX_DEVICE_INFO_BOARD_TYPE, &dev_name);
             }
+            else if (0 == strncmp(key, "WIFIoADDR", strlen(key)))
+            {
+                error = nyx_device_info_query(device, NYX_DEVICE_INFO_WIFI_ADDR, &dev_name);
+            }
+            else if (0 == strncmp(key, "BToADDR", strlen(key)))
+            {
+                error = nyx_device_info_query(device, NYX_DEVICE_INFO_BT_ADDR, &dev_name);
+            }
+
             if (NYX_ERROR_NONE == error)
             {
                 *jstr = g_strdup(dev_name);
@@ -1038,6 +1048,8 @@ static const char* g_non_tokens[] = {
     ,PROP_NAME_FREESPACE
     ,PROP_NAME_PREVPANIC
     ,PROP_NAME_PREVSHUTCLEAN
+    ,PROP_NAME_WIFIOADDR
+    ,PROP_NAME_BTOADDR
 };
 
 static char*
@@ -1118,6 +1130,10 @@ LPSystemCopyStringValue( const char* key, char** jstr )
             err = figurePrevPanic( jstr );
         } else if ( ! strcmp( token, PROP_NAME_PREVSHUTCLEAN ) ) {
             err = figureShutdownClean( jstr );
+        } else if ( ! strcmp( token, PROP_NAME_WIFIOADDR ) ) {
+            err = read_machine_type(jstr, "WIFIoADDR");
+        } else if ( ! strcmp( token, PROP_NAME_BTOADDR ) ) {
+            err = read_machine_type(jstr, "BToADDR");
         } else if ( NULL != (path = getTokenPath( token, TOKENS_DIR )) ) {
             err = readFromFile( path, jstr );
         } else if ( NULL != (path = getTokenPath( token, LP_RUNTIME_DIR )) ) {
-- 
1.8.3.2

